include/common/aarch64/el3_common_macros.S

```diff
@@ -278,8 +278,8 @@
 		 * an earlier boot loader stage.
 		 * -------------------------------------------------------------
 		 */
-		adr	x0, __RW_START__
-		adr	x1, __RW_END__
+		ldr	x0, =__RW_START__
+		ldr	x1, =__RW_END__
```

Comment:
What's the rationale for this change? Using adr is preferred where possible. If this really is needed, it will need to be in a separate commit with explanation.

Reply:
For the adr instruction, it require the lable's offset from the address of this instruction must be in the range +/-1MB. If the option "BL2_IN_XIP_MEM" is set to '1', in some cases, BL2's RW memory will not in the range of +/-1MB from BL2's RO memory region. so we need to use ldr instruction to cover this case.

Reply:
OK, makes sense - thanks.
minor: Could you just fix the spelling in the commit message (limiation and lable)?